adobe/react-spectrum
# Create eslint rule suggesting useLayoutEffect from @react-aria

This PR adds an esLint rule for using the @react-aria/utils implementation of useLayoutEffect, and updates any existing instances in the codebase not using it.

Closes 
#1494 

## ✅ Pull Request Checklist:


## 📝 Test Instructions:


1. Add this import anywhere in the project: ```import { useLayoutEffect } from react```
2. Run ```yarn lint```
3. See that an error is thrown ```error  Please use useLayoutEffect from @react-aria/utils instead  rulesdir/useLayoutEffectRule```

Confirmed all tests are passing after updating existing useLayoutEffect imports:
<img width="353" alt="Screen Shot 2021-12-10 at 10 29 17 AM" src="https://user-images.githubusercontent.com/54876035/145623761-408c0143-367a-4326-8095-c80d3f34d8dc.png">


## 🧢 Your Project:
